bundles/org.eclipse.e4.ui.workbench.renderers.swt/.settings/.api_filters

```diff
@@ -8,4 +8,12 @@
             </message_arguments>
         </filter>
     </resource>
+    <resource path="src/org/eclipse/e4/ui/internal/workbench/swt/OnboardingCTabFolder.java" type="org.eclipse.e4.ui.internal.workbench.swt.OnboardingCTabFolder">
```

Comment:
Why is it necessary? This is internal code so API tools should not run here.

Reply:
I got an API warning in the IDE. I must not extend CTabFolder (it has @noextend annotation).
Eclipse proposed a quick fix which added the entry in the .api_filter file.

The file was already there and it contained an entry already. So the quick fix sounded it reasonable.

Do you think I should remove this entry?

Reply:
@marcushoepfner you probably could *wrap* a CTab control instead of *extending* it and using PaintListener/ControlListenr/... instead? I uses this to show a nice image instead of the gray background for empty editor area.

Reply:
Where do you use this? Do you have an example?
Simplicity is always welcome :)

Reply:
```
CTabFolder cTabFolder = ....

control.addPaintListener(new PaintListener() {

	@Override
	public void paintControl(PaintEvent e) {

		Control[] children = cTabFolder.getTabList();
		if(children.length < 4) {		
			GC gc = e.gc;
			Color background = gc.getBackground();
			Color foreground = gc.getForeground();
			Point size = control.getSize();
			try {
				paintBackground(gc, background, size, bgImage);
			} finally {
				gc.setBackground(background);
				gc.setForeground(foreground);
			}
		}
  	}
}
```

Reply:
I would argue here following: the warning is not there to be ignored, but to re-think the approach. Instead of **extending** CTabFolder (which is bad) you could use ordinary CTabFolder with a listener attached.

So please undo the API change and please try to use default CTabFolder with resize listener.